crates/nu-cli/src/cli.rs

```diff
@@ -683,29 +683,29 @@ pub fn set_rustyline_configuration() -> (Editor<Helper>, IndexMap<String, Value>
             // if the line_editor config section doesn't exist, let's set some defaults
             rl.set_max_history_size(1000);
             rl.set_history_ignore_dups(true);
-            rl.set_history_ignore_space(true);
+            rl.set_history_ignore_space(false);
             rl.set_completion_type(DEFAULT_COMPLETION_MODE);
-            rl.set_completion_prompt_limit(1);
+            rl.set_completion_prompt_limit(100);
             rl.set_keyseq_timeout(500);
-            rl.set_edit_mode(rustyline::config::EditMode::Vi);
-            rl.set_auto_add_history(true);
+            rl.set_edit_mode(rustyline::config::EditMode::Emacs);
+            rl.set_auto_add_history(false);
             rl.set_bell_style(rustyline::config::BellStyle::None);
             rl.set_color_mode(rustyline::ColorMode::Enabled);
-            rl.set_tab_stop(4);
+            rl.set_tab_stop(8);
         }
     } else {
         // if the config file itself doesn't exist, let's set some defaults
         rl.set_max_history_size(1000);
         rl.set_history_ignore_dups(true);
-        rl.set_history_ignore_space(true);
+        rl.set_history_ignore_space(false);
         rl.set_completion_type(DEFAULT_COMPLETION_MODE);
-        rl.set_completion_prompt_limit(1);
+        rl.set_completion_prompt_limit(100);
         rl.set_keyseq_timeout(500);
-        rl.set_edit_mode(rustyline::config::EditMode::Vi);
-        rl.set_auto_add_history(true);
+        rl.set_edit_mode(rustyline::config::EditMode::Emacs);
+        rl.set_auto_add_history(false);
         rl.set_bell_style(rustyline::config::BellStyle::None);
         rl.set_color_mode(rustyline::ColorMode::Enabled);
-        rl.set_tab_stop(4);
+        rl.set_tab_stop(8);
```

Comment:
There's a lot of duplication here, and if we're going for using rustyline's defaults, I think a better way to write this would be:

```rust
let mut rl = config::config(Tag::unknown())
    .ok()
    .and_then(|config| config.get("line_editor"))
    .map(|line_editor_vars| {
        let mut rl: Editor<_> = Editor::with_config(config);

        for (idx, value) in line_editor_vars.row_entries() {
            ...
        }
    })
    .unwrap_or_default();

rl.bind_sequence(...);
...
```

I _think_ something like that should work. What do you think?

If we would rather not rely on something externally for the defaults, we could take the same approach, but instead of `unwrap_or_default` we could `unwrap_or_else` and set the defaults in the block.

Reply:
I was planning on taking those 2 sections of defaults out and moving it to the front of the function. Then if there are items set, they'll override those initial defaults. This should remove the duplication.

Reply:
That works too 🚀